src/internal/analytics-metadata/__tests__/dom-utils.test.tsx

```diff
@@ -0,0 +1,106 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React from 'react';
+import { render } from '@testing-library/react';
+import { getAnalyticsMetadataAttribute, METADATA_ATTRIBUTE } from '../attributes';
+import { findNextNode, isNodeComponent, findComponentUp } from '../dom-utils';
+
+describe('findNextNode', () => {
+  test('finds parent', () => {
+    const { container } = render(
+      <div id="parent">
+        <div id="child"></div>
+      </div>
+    );
+    const child = container.querySelector('#child');
+    expect(findNextNode(child as HTMLElement)?.id).toEqual('parent');
+  });
+  test('returns null when child does not exist', () => {
+    const { container } = render(
+      <div id="parent">
+        <div id="child"></div>
+      </div>
+    );
+    const child = container.querySelector('#wrong-child');
+    expect(findNextNode(child as HTMLElement)).toBeNull();
+  });
+  test('returns null when parent does not exist', () => {
+    const { container } = render(
+      <div id="parent">
+        <div id="child"></div>
+      </div>
+    );
+    expect(findNextNode(container.parentElement?.parentElement as HTMLElement)).toBeNull();
+  });
+  test('returns element referred to with referrerId', () => {
+    const { container } = render(
+      <div>
+        <div id="child" data-awsui-referrer-id="id:rr5:"></div>
+        <div id="id:rr5:" className="parent"></div>
+      </div>
+    );
+    const child = container.querySelector('#child');
+    expect(findNextNode(child as HTMLElement)?.classList.contains('parent')).toBe(true);
+  });
+  test('returns null when element referred to with referrerId does not exist', () => {
```

Comment:
Curious about this one. If the referrer doesn't exist then technically the next "parent" could follow the normal traversal up.

Reply:
if there is a referrer id that does not refer to anything there is an issue in the interface, so I don't want to ignore that